sycl/source/detail/error_handling/enqueue_kernel.cpp

```diff
@@ -21,18 +22,40 @@ namespace detail {
 
 namespace enqueue_kernel_launch {
 
-bool oclHandleInvalidWorkGroupSize(const device_impl &DeviceImpl,
-                                   pi_kernel Kernel, const NDRDescT &NDRDesc) {
+bool handleInvalidWorkGroupSize(const device_impl &DeviceImpl, pi_kernel Kernel,
+                                const NDRDescT &NDRDesc) {
   const bool HasLocalSize = (NDRDesc.LocalSize[0] != 0);
 
   const plugin &Plugin = DeviceImpl.getPlugin();
   RT::PiDevice Device = DeviceImpl.getHandleRef();
+  cl::sycl::platform Platform = DeviceImpl.get_platform();
+
+  if (HasLocalSize) {
+    size_t MaxThreadsPerBlock[3] = {};
+    Plugin.call<PiApiKind::piDeviceGetInfo>(
+        Device, PI_DEVICE_INFO_MAX_WORK_ITEM_SIZES, sizeof(MaxThreadsPerBlock),
+        MaxThreadsPerBlock, nullptr);
+
+    for (size_t I = 0; I < 3; ++I) {
+      if (MaxThreadsPerBlock[I] < NDRDesc.LocalSize[I]) {
+        throw sycl::nd_range_error(
+            "The number of work-items in each dimension of a work-group cannot "
+            "exceed info::device::max_work_item_sizes which is {" +
```

Comment:
```suggestion
            "The number of work-items in each dimension of a work-group cannot "
            "exceed {" +
```
I think it's better to avoid using API names in error message to simplify maintenance.